src/plugins/network/network.c

```diff
@@ -50,6 +52,75 @@ int elektraNetworkAddrInfo (Key * toCheck)
 	return 0;
 }
 
+int elektraPortInfo (Key * toCheck, Key * parentKey)
+{
+	const Key * meta = keyGetMeta (toCheck, "check/port");
+	const Key * listenMeta = keyGetMeta (toCheck, "check/port/listen");
+	if (!meta && !listenMeta) return 0; /* No check to do */
+	char * endptr = NULL;
+	long portNumber = strtol (keyString (toCheck), &endptr, 10);
+	int portNumberNetworkByteOrder;
+
+	if (*endptr == '\0')
+	{
+		if (portNumber < 0 || portNumber > 65535)
+		{
+			ELEKTRA_SET_ERRORF (201, parentKey, "Port %ld on key %s was not within 0 - 65535", portNumber, keyName (toCheck));
+			return -1;
+		}
+		portNumberNetworkByteOrder = htons (portNumber);
+	}
+	else
+	{
+		struct servent * service;
+		service = getservbyname (keyString (toCheck), NULL); // NULL means we accept both tcp and udp
+		if (service == NULL)
+		{
+			ELEKTRA_SET_ERRORF (202, parentKey, "Could not find service with name %s on key %s", keyString (toCheck),
+					    keyName (toCheck));
+			return -1;
+		}
+		portNumberNetworkByteOrder = service->s_port;
+	}
+
+	// Check if we can connect to it
```

Comment:
comment?

Reply:
The code which follows basically does this. I will just move the comment a few lines below so it is more clear